lintreview/tools/phpcs.py

```diff
@@ -36,11 +36,20 @@ def process_files(self, files):
         to save resources.
         """
         log.debug('Processing %s files with %s', files, self.name)
+
+        if self.options.get('install'):
+            output = run_command(
+                ['composer', 'install'],
+                ignore_error=True,
+                cwd=self.get_working_dir())
```

Comment:
I have similar concerns with PHP dependencies as I have with running `npm` install around time and cost.

Reply:
Do you often find that your PHP applications require additional composer packages to install their PHPCS rules?

Reply:
As with npm, it's optional.

As for additional composer packages, I'm not really familiar with PHPCS. Just wanted to demonstrate on a non-node linter. The point is that a team could add a PHP project to their existing projects without having to touch the linting bot.

Reply:
I have plans in the near future to extract the various tool runtime environments into small standalone docker images removing the need for a kitchen sink image entirely. I feel this change would not be necessary if phpcs was a separate image people could pull down as needed.

I would prefer to not add features that will be removed in the short term.

Reply:
Are you saying the linting bot would start docker containers for linting environments? Or that you would have different "flavors" of the linting bot docker image, each with different tools?

Reply:
That the review jobs would spawn containers to run each tool on the repository being reviewed.

Reply:
Awesome! That's actually how I initially thought I was going to solve my problem when I started this work, but when I saw how easy it was to get the functionality I needed by just re-jiggering a few lines...

Spawning containers might solve some problems, but wouldn't you still need to have all your eslint config and plugin packages for all your projects pre-installed in the linter docker image? And update the image when your config changes? I also learned that this does not only apply to eslint. With PHPCS, for example, you can add new "sniffs" as composer packages.

Reply:
> you still need to have all your eslint config and plugin packages for all your projects pre-installed in the linter docker image? 

Yes you would. It doesn't help address the 'unresolved imports' for eslint. But it would allow lintreview users to build their own docker images that contain the custom sniffs/packages they use.

Reply:
I think that would be great for a lot of people. However, I also think many would much rather not have to maintain separate docker images. I'm already doing the work of keeping track of needed dependencies when I update my package.json/composer.json/etc.. I wouldn't want to then have to go and update another tool with those same changes.

Reply:
I hear you on the pain of setting up docker images. For the node based projects I think the `install` option is one of the only ways to get around the unresolved imports due to how babel & eslint handle those checks. However, with PHP and other toolchains linting is far less invasive and changes far less frequently than in Javascript projects.